### PR TITLE
App layout overhaul

### PR DESCRIPTION
**Pull request by mgramigna**
~Draft for now, review #31 and #33 first~

# Summary

Updates layout/styling of the app to adhere to the recent mockups that we received. Completely re-does the file structure. Here is a useful diagram of the new structure for reference:

image

image

## New Behavior

* Patient creation/viewing is scoped to a patient resource now. Calculation/FHIR resource addition/editing happens elsewhere
* Data elements are now in a searchable dropdown box to be selected. Code editing is the same
* Calculation is done via a button that lives in the rightmost panel

## Code Changes

* Overhaul file structure, mainly the `components` directory
  * Updated `__tests__` to adhere to this structure as well

image

* Extract small/purposeful code into individual components (each panel, each piece of info within each panel, etc)
* Create/modify all the styling to adhere to the mockups
* Small styling updates (button colors, modal size)
* Add select component that replaces data element selection and searching (not worth the overhead of using Fuse anymore since search is natively supported by Mantine `Select`
  * Introduce a new recoil atom to hold a lookup of the values of this select box (strings) to full data requirement objects
    * This was necessary because we cannot pass objects as props to each select item and handle it in the `onChange`. We need to use a string for the lookup (ping me if you need a better explanation lol)
* Update/fix all tests
* Added mask to clause coverage route to hide html from URL

# Testing Guidance

Test literally everything. Honestly, I think looking through code diff won't be too helpful due to the renaming/moving of everything. Feel free to look at code diffs of things that didn't exist prior to this PR though.

### CODE REVIEW

**Review by elsaperelli — COMMENTED** [review]
The app looks amazing and the renaming and restructuring was really necessary. Just a few comments and questions but looks great!!!

**Review by sarahmcdougall — CHANGES_REQUESTED** [review]
Aside from in-line comments, here are a few thoughts that came up while reviewing (that don’t necessarily require action but I thought I’d include them here):
* Are there any relevant unit tests that should be added for DataRequirementSelectOption.tsx, PatientInfocard.tsx, or ResourceInfoCard.tsx? I don’t think we have test files for them currently
* I know that we don’t need the SearchBar for searching through resources anymore, so I've been thinking about whether there would be anything (currently implemented or in the future) that should be searchable. I was wondering if it would be helpful for the patients to be searchable by patient name? I’ve been testing with a small number of patients, so I’m not sure how useful it would be, but it’s something we can think about.

File: components/TestCaseEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  const { classes } = useStyles();
+
+  const renderPanelPlaceholderText = () => {
+    return (
+      <Text color="dimmed">
+        <i>Select a patient to add resources and see more information</i>
+      </Text>
+    );
+  };
```
[the remark this answers is not in the document]
Comment (mgramigna) [reply components/TestCaseEditor.tsx:32]: Discussed/resolved through slack

File: components/utils/PatientInfoCard.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      withBorder
+      sx={theme => {
+        const style: Sx = {
+          ':hover': {
+            cursor: 'pointer',
+            backgroundColor: theme.colors.gray[0]
+          }
+        };
+        if (selected) {
+          style.borderColor = theme.primaryColor;
```
[the remark this answers is not in the document]
Comment (mgramigna) [reply components/utils/PatientInfoCard.tsx:34]: Yep! I thought `theme.primaryColor` would be that, but turns out we also need `theme.primaryShade` lol

File: util/fhir.ts [hunk 3]
```diff
@@ -73,6 +73,10 @@ export function getPatientInfoString(patient: fhir4.Patient) {
   return `${getPatientNameString(patient)} (DOB: ${patient.birthDate})`;
```
[the remark this answers is not in the document]
Comment (mgramigna) [reply util/fhir.ts]: Addressed in https://github.com/projecttacoma/fqm-testify/pull/34/commits/733b910b37d2833d36152e3c3e1e85863d6595dd

File: components/calculation/MeasureHighlightingPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        });
+      }
+    }
+  };
+
+  // Function to calculate the selected patient's measure report
+  const calculate = async (id: string | null) => {
+    const options: CalculatorTypes.CalculationOptions = {
+      calculateHTML: true,
+      // TODO: Flip this to true once a new fqm-execution version is released/dependency is updated
```
[the remark this answers is not in the document]
Comment (mgramigna) [reply components/calculation/MeasureHighlightingPanel.tsx]: Addressed in https://github.com/projecttacoma/fqm-testify/pull/34/commits/733b910b37d2833d36152e3c3e1e85863d6595dd

File: components/TestCaseEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import { createStyles, Grid, Text } from '@mantine/core';
+import React from 'react';
+import { useRecoilValue } from 'recoil';
+import { selectedPatientState } from '../state/atoms/selectedPatient';
+import PatientCreationPanel from './patient-creation/PatientCreationPanel';
+import ResourcePanel from './resource-creation/ResourcePanel';
+import MeasureHighlightingPanel from './calculation/MeasureHighlightingPanel';
+
+const useStyles = createStyles({
+  highlightedMarkup: {
```
[the remark this answers is not in the document]
Comment (mgramigna) [reply components/TestCaseEditor.tsx]: Addressed in https://github.com/projecttacoma/fqm-testify/pull/34/commits/733b910b37d2833d36152e3c3e1e85863d6595dd

File: __tests__/components/utils/AppHeader.test.tsx [hunk 6]
```diff
@@ -1,11 +1,11 @@
 import { render, screen } from '@testing-library/react';
 import '@testing-library/jest-dom';
-import { mantineRecoilWrap } from '../helpers/testHelpers';
-import AbacusHeader from '../../components/AbacusHeader';
+import { mantineRecoilWrap } from '../../helpers/testHelpers';
+import AppHeader from '../../../components/utils/AppHeader';
 
 describe('AbacusHeader', () => {
```
[the remark this answers is not in the document]
Comment (mgramigna) [reply __tests__/components/utils/AppHeader.test.tsx]: Addressed in https://github.com/projecttacoma/fqm-testify/pull/34/commits/733b910b37d2833d36152e3c3e1e85863d6595dd

File: components/calculation/MeasureHighlightingPanel.tsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+      calculateSDEs: false,
```
[the remark this answers is not in the document]
Comment (mgramigna) [reply components/calculation/MeasureHighlightingPanel.tsx]: Addressed in https://github.com/projecttacoma/fqm-testify/pull/34/commits/733b910b37d2833d36152e3c3e1e85863d6595dd

File: components/resource-creation/ResourcePanel.tsx [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+  const renderPanelPlaceholderText = () => {
+    return (
+      <Text color="dimmed">
+        <i>Select a patient to add resources and see more information</i>
+      </Text>
+    );
+  };
+
+  const renderDataElements = () => {
```
[the remark this answers is not in the document]
Comment (mgramigna) [reply components/resource-creation/ResourcePanel.tsx]: Renamed to a better function name in https://github.com/projecttacoma/fqm-testify/pull/34/commits/74c0b32613a3fe42a0d434ec01a8758d6d815c53

File: components/utils/PatientInfoCard.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mgramigna) [reply components/utils/PatientInfoCard.tsx]: Addressed in https://github.com/projecttacoma/fqm-testify/commit/733b910b37d2833d36152e3c3e1e85863d6595dd

File: components/patient-creation/PatientCreationPanel.tsx [hunk 9]
```diff
@@ -39,6 +45,78 @@ export default function ResourceCreationPanel() {
...
+  const getConfirmationModalText = (patientId: string | null) => {
+    let patientName;
+    if (patientId !== null) {
+      const patient = currentPatients[patientId].patient;
+      patientName = getPatientNameString(patient);
+    }
+    return `Are you sure you want to delete ${patientName || 'this patient'}?`;
+  };
+
   const exportAllPatients = () => {
```
[the remark this answers is not in the document]
Comment (mgramigna) [reply components/patient-creation/PatientCreationPanel.tsx:120]: I think that's right. Eventually, this component will only be rendered IFF a measure bundle exists (both name and content)

File: __tests__/components/patient-creation/PatientCreationPanel.test.tsx [hunk 10]
```diff
@@ -56,20 +27,29 @@ describe('PatientCreation', () => {
...
+      name: /create/i
+    });
+
+    expect(createButton).toBeInTheDocument();
+
+    await act(async () => {
+      fireEvent.click(createButton);
+    });
 
+    const modal = screen.getByTestId('code-editor-modal');
```
[the remark this answers is not in the document]
Comment (mgramigna) [reply __tests__/components/patient-creation/PatientCreationPanel.test.tsx]: Resolved in https://github.com/projecttacoma/fqm-testify/pull/34/commits/c7422da857fe14f7d9d5a89e4ff03fc845dbce97

**Review by elsaperelli — APPROVED** [review]
Looks good to me!!

File: __tests__/components/utils/ResourceInfoCard.test.tsx [hunk 11]
```diff
@@ -0,0 +1,88 @@
+import '@testing-library/jest-dom';
+import { fireEvent, render, screen } from '@testing-library/react';
+import ResourceInfoCard, { ResourceInfoCardProps } from '../../../components/utils/ResourceInfoCard';
+import { mantineRecoilWrap } from '../../helpers/testHelpers';
+
+const EXAMPLE_RESOURCE_TYPE = 'Procedure';
+const EXAMPLE_LABEL = 'Colonoscopy (http://example.com/ValueSet/example-vs';
```
Comment (sarahmcdougall) [inline __tests__/components/utils/ResourceInfoCard.test.tsx:7]:
```suggestion
const EXAMPLE_LABEL = 'Colonoscopy (http://example.com/ValueSet/example-vs)';
```

**Review by sarahmcdougall — CHANGES_REQUESTED** [review]

**Review by sarahmcdougall — APPROVED** [review]

**Pull request merged**